src/common/Set/Set.tsx

```diff
@@ -0,0 +1,69 @@
+import React from "react";
+import { useState, useEffect } from "react";
+import { Card } from "../Card";
+import { Section } from "./interfaces";
+import * as UI from "./style";
+import PropTypes from "prop-types";
+
+/**
+ * Set component
+ */
+export const Set: React.FC<Section> = ({ info }) => {
+  const [eventList, setEventList] = useState(info.eventList.slice(0, 4));
+  const [showMore, setShowMore] = useState(false);
+  const [btnText, setBtnText] = useState("Show More");
+
+  /* Updates the set component if when the show more status has been changed */
+  useEffect(() => {
+    if (showMore) {
+      setEventList(info.eventList);
+      setBtnText("Show Less");
+    } else {
+      setEventList(info.eventList.slice(0, 4));
+      setBtnText("Show More");
+    }
+  }, [showMore]);
```

Comment:
You probably don't need this extra hook here. You should be able to add logic to the `toggleEventDisplay` function that achieves this. I tested it out, this was my code for the function:
```javascript
const toggleEventsDisplay = () => {
    if (showMore) {
      setEventList(info.eventList.slice(0, 4));
      setBtnText("Show More");
    } else {
      setEventList(info.eventList);
      setBtnText("Show Less");
    }
    setShowMore(!showMore);
};
```

Reply:
Nah I actually like the current implementation better. Putting things in the callback function could confuse things in the future